valhalla/baldr/traffictile.h

```diff
@@ -0,0 +1,129 @@
+#ifndef VALHALLA_BALDR_TRAFFICTILE_H_
+#define VALHALLA_BALDR_TRAFFICTILE_H_
+
+#include <cassert>
+#include <cstdint>
+#include <vector>
+
+namespace valhalla {
+namespace baldr {
+namespace traffic {
+struct Speed {
+  std::uint16_t speed_kmh : 7;        // km/h - so max range is 0-127km/h
+  std::uint16_t congestion_level : 3; // some value from 0 to 7 to report back
+  std::uint16_t is_scale : 1;         // treat speed as a floating point multiplier to edge speed
+  std::uint16_t age : 4;              //
+  std::uint16_t has_incident : 1;     // TODO: reserved for whether edge has an incident or not
+};
+
+// single incident record
+struct Incident {
+  std::uint64_t edge_index : 21;
+  std::uint64_t incident_type : 8;
+  std::uint64_t start_location : 10;
+  std::uint64_t length : 10;
+  std::uint64_t spare : 15;
+  Incident(const Incident& other) = default;
+  Incident(const volatile Incident& other)
+      : edge_index{other.edge_index}, incident_type{other.incident_type},
+        start_location{other.start_location}, length{other.length}, spare{0} {
+  }
+  Incident& operator=(const Incident& other) = default;
+};
+
+// per-speed-tile header
+struct TileHeader {
+  std::uint64_t tile_id;
+  std::uint32_t directededge_count;
+  std::uint32_t incident_buffer_size;
+  std::uint64_t active_incident_buffer : 1;
```

Comment:
is this whether or not we should expect an incidents header after the speed/congestion data? what advantage does this have over just always having the header but having it there? i know its a draft, not asking for changes, just trying to understand

Reply:
After brainstorming with you and @PureW , I decided to throw in a simple double-buffer solution for handling variable-length incident lists.  Whether this implementation will make it through to the final PR remains to be seen, but it was easy enough to add, so I did.

Reply:
excellent double buffer sounds good